apkid/rules/elf/common.yara

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2018  RedNaga. https://rednaga.io
+ * All rights reserved. Contact: rednaga@protonmail.com
+ *
+ *
+ * This file is part of APKiD
+ *
+ *
+ * Commercial License Usage
+ * ------------------------
+ * Licensees holding valid commercial APKiD licenses may use this file
+ * in accordance with the commercial license agreement provided with the
+ * Software or, alternatively, in accordance with the terms contained in
+ * a written agreement between you and RedNaga.
+ *
+ *
+ * GNU General Public License Usage
+ * --------------------------------
+ * Alternatively, this file may be used under the terms of the GNU General
+ * Public License version 3.0 as published by the Free Software Foundation
+ * and appearing in the file LICENSE.GPL included in the packaging of this
+ * file. Please visit http://www.gnu.org/copyleft/gpl.html and review the
+ * information to ensure the GNU General Public License version 3.0
+ * requirements will be met.
+ *
+ **/
+
+private rule is_elf
+{
+    meta:
+        description = "Identifies an ELF binary based on its magic number"
+        author      = "Eduardo Novella"
+
+    strings:
+        $elf_magic = { 7F 45 4C 46 }
```

Comment:
I think this should be replaced with something that's always true if it's an elf:

```
    condition:
        elf.number_of_sections >= 0
```

This way the elf module is responsible for knowing elf magic, and we're not duplicating that responsibility. I didn't know about this technique when we made `is_dex`, but that function could also be replaced with something similar.

Reply:
Modified and works okay.